lib/app/service/installer.go

```diff
@@ -381,14 +390,37 @@ main "$@"
 #
 # Script for upgrading the currently running application to a new version.
 #
+# Copyright 2016 Gravitational, Inc.
+#
+# This file is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
 
 if [[ $(id -u) -ne 0 ]]; then
   echo "please run this script as root" && exit 1
 fi
 
 scriptdir=$(dirname $(realpath $0))
-app=$($scriptdir/gravity app-package --state-dir=$scriptdir)
-$scriptdir/upload && $scriptdir/gravity --insecure update trigger $app
+app=$("$scriptdir/gravity" app-package --state-dir="$scriptdir")
+"$scriptdir/upload" && "$scriptdir/gravity" --insecure update trigger $app
+`
+
+	checkScript = `#!/bin/bash
+#
+# Script for executing preflight checks.
```

Comment:
> First, it's very easy to fire up the checks just by invoking the script, without remembering the command / trying to figure out what flags you might need. And the command is there for you for more involved scenarios.
Well this could be embedded in the logic of the binary itself. If it were me, for the pre-install checks, I would even add the ability to ask the user which profile they want to use. If only a single profile is defined, just use it, if someone wants to automate it accept the flag, and if someone doesn't know, give a numbered list that they can select from. Although I would do this on the installer as well.

> Also, it serves a sort of "educational" purpose for the user, i.e. like you said we don't really advertise anywhere in the installer that you can run preflight checks before launching operation while seeing the script will make them realize they can do that. Maybe I should call it something more prominent though like run_preflight_checks rather than generic check.
I guess my line of thinking is why do we want to educate someone on this. If I see a script called `checks` or a script called `run_preflight_checks` as someones first interaction with a new product, these things mean nothing to me. 

Where as someone asking for the ability to run pre-flight checks manually, so it creates less work for them when the checks would fail, seems to point to a UX problem in the upgrade itself. That this is a common pain point in the upgrade, that checks will fail, and then someone is left with a partial upgrade to clean up, especially if rooted in a bug in the checks themselves as recently occurred. 

To me the only reason to introduce a new UX pattern specifically around checks is:
- If I want to run upgrade checks using the new version of the gravity binary, but don't want to actually do an upgrade, or make the decision on continuing with an upgrade separately. Is this is what's being solved with this pattern?
- I want the ability to trigger upgrade checks outside of doing an actual upgrade (IE to check if things are healthy). Our UX pattern for this is already to use `gravity status`.
- I want the ability as a developer to test a new version of checks without as much hassle (Something like this probably shouldn't be at the forefront).

Reply:
> If I want to run upgrade checks using the new version of the gravity binary, but don't want to actually do an upgrade, or make the decision on continuing with an upgrade separately. Is this is what's being solved with this pattern?

Yep. It was requested by a customer.